### ISSUE TITLE
Use info from `ApplicationStartInfo` API on Android 15 

### ISSUE THREAD

**Issue by yogurtearl** [getsentry/sentry-java#3519]
### Problem Statement

Would like to have info provided in `ApplicationStartInfo` API on Android 15 available thru Sentry. 

see https://developer.android.com/about/versions/15/features#app-start-info

### Solution Brainstorm

You can use this with `if( SDK >= 35 )`  and still get useful data about application start up metrics for device running Android 15+.

**Comment by kahest** [comment]
Thanks @yogurtearl for the feature request - we discussed this (and other features based on new Android 15 APIs) and will likely implement this at some point.

**Comment by AndroidDeveloperLB** [comment]
How do you use ApplicationStartInfo ? I can't see any docs for how to get an instance of it

**Comment by markushi** [comment]
@AndroidDeveloperLB The `ApplicationStartInfo` can be received by registering a ActivityManager.addApplicationStartInfoCompletionListener hook, [see the docs](https://developer.android.com/reference/android/app/ActivityManager#addApplicationStartInfoCompletionListener(java.util.concurrent.Executor,%20java.util.function.Consumer%3Candroid.app.ApplicationStartInfo%3E)) for more details.

**Comment by AndroidDeveloperLB** [comment]
@markushi Thank you!
I can see detection of the reason of starting the app, but I can't find which app started it (when it's relevant). Is it possible?

**Comment by markushi** [comment]
@AndroidDeveloperLB I think what you're looking for is `Activity.getReferrer`([link](https://developer.android.com/reference/android/app/Activity.html#getReferrer())). In which situations would this be useful for you?

**Comment by AndroidDeveloperLB** [comment]
@markushi Seems you are correct, but need to add visibility to the apps. I wonder if a simple launchable filter will be fine:

```
<intent>
    <action android:name="android.intent.action.MAIN" />
    <category android:name="android.intent.category.LAUNCHER" />
</intent>
```

I think for most cases it should suffice. Anyway, thank you!

**Comment by romtsn** [comment]
fwiw you can also retrieve it via [getHistoricalProcessStartReasons](https://developer.android.com/reference/android/app/ActivityManager#getHistoricalProcessStartReasons(int)), which would only return them for your app.

**Comment by AndroidDeveloperLB** [comment]
@romtsn What do you mean? What is "only return the app start infos for your app" ? 

What's the difference between this and addApplicationStartInfoCompletionListener , if I check only the first item in the returned list? Or maybe it's for the previous starts of the app?

**Comment by kahest** [comment]
We'll investigate this more closely wrt which APIs make sense to use and what additional data they can provide. We'll follow up here

**Comment by romtsn** [comment]
From @markushi 

> Let's add the ApplicationStartInfo as part of the [App context](https://github.com/getsentry/sentry-java/blob/main/sentry/src/main/java/io/sentry/protocol/App.java). We also need to make the new attributes indexable, so we can filter and search for them.

**Comment by Angelodaniel** [comment]
Seems the [[DEPRECATED] Mobile SDKs](https://github.com/orgs/getsentry/projects/19) is deprecated, are still things on the backlog? @romtsn

**Comment by stefanosiano** [comment]
@Angelodaniel yep, the project just moved, but things are still on the backlog
